### PR TITLE
GuideRateConfig object

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by joakim-hove** [comment]
@totto82 : This still has some way to go - but it would be good if you could verify that the API makes sense. The plan is that the simulator will first instantiate - and manage - a `GuideRate` instance:
```
GuideRate guideRate( schedule );
```
then the simulator will ask this object for a guide rate value:
```
double guide_rate = guideRate.update("WELL_OR_GROUP", time_step, oil_pot, gas_pot, wat_pot);
```

Does seem like a good plan?

File: src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    guide_rate = group.guide_rate;
+    dyn_state.update( report_step, GuideRateValue{sim_time, guide_rate} );
+}
+
+
+void GuideRate::well_update(const std::string& wgname, size_t report_step, double oil_pot, double gas_pot, double wat_pot) {
+    const auto& config = this->schedule.guideRateConfig(report_step);
+    auto sim_time = this->schedule.getTimeMap().seconds(report_step);
+    const auto& well = config.well(wgname);
```
Comment (totto82) [inline src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp]: The well_update is called every time_step not only per report_step. Therefore I think we need to pass the actual time. (Same for groups)

File: opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.hpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+private:
+    void well_update(const std::string& wgname, size_t report_step, double oil_pot, double gas_pot, double wat_pot);
+    void group_update(const std::string& wgname, size_t report_step, double oil_pot, double gas_pot, double wat_pot);
+    double get(const std::string& wgname, size_t report_step) const;
+    double eval(const GuideRateModel& model, const GuideRateValue& prev) const;
+    double eval_group_pot() const;
+    double eval_group_resvinj() const;
+
+    std::unordered_map<std::string,DynamicState<GuideRateValue>> values;
+    const Schedule& schedule;
```
[the remark this answers is not in the document]
Comment (joakim-hove) [reply opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.hpp]: Fair point

File: src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp]: OK - good; then I guess we will need both the report_step and the actual time.

**Comment by joakim-hove** [comment]
OK - this should be ready to be merged now; it is not ready as in complete, but @totto82 and myself has agreed it can be merged; and then worked on further. It is not in use yet.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by totto82** [comment]
Thanks. I will test it in combination with new control code.

**Pull request merged**